client/my-sites/site-settings/seo-settings/help.jsx

```diff
@@ -25,16 +26,18 @@ const SeoSettingsHelpCard = ( {
 	hasAdvancedSEOFeature,
 	siteId,
 	siteIsJetpack,
+	siteIsFreeAtomic,
 	translate,
 } ) => {
-	const seoHelpLink = siteIsJetpack
-		? 'https://jetpack.com/support/seo-tools/'
-		: 'https://wpbizseo.wordpress.com/';
+	const seoHelpLink =
+		siteIsJetpack && ! siteIsFreeAtomic
+			? 'https://jetpack.com/support/seo-tools/'
+			: 'https://wpbizseo.wordpress.com/';
```

Comment:
Checking for `isFreeAtomic` here as well, to keep Atomic access to Jetpack features.